kip-0017.md

```diff
@@ -8,105 +8,426 @@ Category: Interface
 
-These are valid kadena chain-ids
+### Example Pairing Proposal Request and Settlement Response
 
+```jsonc
+// Proposal request
+{
+  // ... other properties from the WalletConnect Proposal request
+  "requiredNamespaces": {
+    "kadena": {
+      "chains": ["kadena:mainnet01", "kadena:testnet04", "kadena:development"],
+      "methods": [
+        "kadena_get_accounts_v1",
+        "kadena_sign_v1",
+        "kadena_quicksign_v1"
+      ],
+      "events": []
+    }
+  }
+}
 ```
-# Kadena mainnet
-kadena:mainnet01
 
-# Current kadena testnet
-kadena:testnet04
+```jsonc
+// Settlement response
+{
+  // ... other properties from the WalletConnect Settlement response
+  "namespaces": {
+    "kadena": {
+      "accounts": [
+        "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+        "kadena:testnet04:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+        "kadena:testnet04:22ddc64851718e9d41d98b0f33d5e328ae5bbbbd97aed9885adac0f2d070ff9c"
+      ],
+      "methods": [
+        "kadena_get_accounts_v1",
+        "kadena_sign_v1",
+        "kadena_quicksign_v1"
+      ],
+      "events": []
+    }
+  }
+}
+```
+
+### WalletConnect Proposal Request
+
+WalletConnect dictates a `requiredNamespaces` property, that contains:
 
-# Kadena devnet
-kadena:development
+1. `requiredNamespaces`: `Object` 
+   1.1. `chains`: `Array<string>` - an array of [Chain IDs](#wallet-connect-chain-ids)
+   1.2. `methods`: `Array<string>` - an array of [Methods](#walletconnect-methods) 
+   1.3. `events`: `Array<string>` - an array of [Events](#walletconnect-events)
+
+### WalletConnect Settlement Response
+
+The response for the Proposal is an object that contains the following properties:
+
+1. `namespaces`: `Object`
+    1.1. `accounts`: `Array<string>` - an array of [Accounts](#wallet-connect-accounts) availble for signing.
+    1.2. `methods`: `Array<string>` - an array of [Methods](#walletconnect-methods) available.
+    1.3. `events`: `Array<string>` - an array of [Events](#walletconnect-events) available.
+
+## Wallet Connect Chain IDs
+
+Chain IDs are Wallet Connect’s way of identifying specific blockchains within an ecosystem.
+
+They are defined by the [CAIP-2 standard](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md), and are formatted as:
+
+```
+    <namespace> + ":" + <reference>
 ```
+- `<namespace>` always refers to "kadena".
+- `<reference>` refers to a unique network ID within the blockchain ecosystem (e.g. testnet, mainnet).
 
-## Accounts
+These are valid Kadena Chain IDs:
+
+- `kadena:mainnet01` - Kadena mainnet
+- `kadena:testnet04` - Current kadena testnet
+- `kadena:development` - Kadena devnet
+
+> **NB**: `Chain IDs`, as defined here, are semantically specific to
+> WalletConnect and is not equivalent to Kadena's "chain IDs".
+> They are more closely related to Kadena's `networkId` field.
+> <Read the motivation for this in TODO>
```

Comment:
Added this TODO in case we wanted to document why we aren't including chain ids in the Wallet Connect chain ID.

Reply:
Added quick explanation that we're providing Kadena chain ID information in the getAccounts method.